src/pages/UserSettings/UserSettings.tsx

```diff
@@ -0,0 +1,209 @@
+      },
+      onSuccess: () => {
+        setChangeUsernameState("Username updated. Refresh to see changes.");
+      },
+    });
+
+  const { isLoading: emailChangeLoading, mutate: changeEmail } = useMutation({
+    mutationFn: (email: string) => {
+      return api.put("/user/info/", {
+        data_type: "email",
+        data: email,
+      });
+    },
+    onSuccess: () => {
+      setChangeEmailState("Email successfully updated");
+    },
+    onError: () => {
+      setChangeEmailState("Email may already be taken.");
+    },
+  });
+
+  return (
+    <>
+      <Column className="mx-8 my-16 w-full gap-6">
+        <h2>Settings</h2>
+        <h4 className="text-gray-600">Notification Settings</h4>
+        <Row className=" h-6 w-1/2 items-center">
+          <p className="text-gray-400">Email Alerts</p>
+          <div className="flex flex-1" />
+          <Toggle
+            toggled={hasEmailAlerts}
+            onToggle={() => {
+              updateEmailAlerts.mutate(!hasEmailAlerts);
+            }}
+          />
+        </Row>
+        {hasEmailAlerts && (
+          <>
+            <TextField
+              value={emailTime}
+              onChangeValue={(emailTime) => {
+                setEmailTimeNetworkingState(Constants.LOADING_MUTATION_TEXT);
+                setEmailTime(emailTime);
+              }}
+              label="Pick a time to receive notifications:"
+              type="time"
+            />
+            <label className="font-normal text-gray-600">
+              {emailTimeNetworkingState}
+            </label>
+          </>
+        )}
+        <h4 className="text-gray-600">Account Settings</h4>
+        <TextField
+          value={email}
+          hint="Enter a new email address"
+          onChangeValue={setEmail}
+          label="Change email:"
+        />
+        {changeEmailState && (
+          <label className="font-normal">{changeEmailState}</label>
+        )}
+        <Button
+          label="Change Email"
+          icon={emailChangeLoading ? <Spinner variaton="SMALL" /> : undefined}
+          onClick={() => {
+            if (EmailValidator.validate(email)) {
+              changeEmail(email);
+            } else {
+              setChangeEmailState("Email not valid");
+            }
+          }}
+        />
+        <TextField
+          value={username}
+          hint="Enter a new username"
+          onChangeValue={setUsername}
+          label="Change username:"
+        />
+        <Button
+          label="Change Username"
+          icon={
+            usernameChangeLoading ? <Spinner variaton="SMALL" /> : undefined
+          }
+          onClick={() => {
+            if (username) {
+              changeUsername(username);
+            } else {
+              setChangeUsernameState("Username must not be empty");
+            }
+          }}
+        />
+        {changeUsernameState && (
+          <label className="font-normal">{changeUsernameState}</label>
```

Comment:
make the error text red? also this should probably be before the change username button right?